tools-local/init-build.proj

```diff
@@ -0,0 +1,323 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <!-- Don't try to import the target info file when building this project -->
+    <GeneratingStaticPropertiesFile>true</GeneratingStaticPropertiesFile>
+  </PropertyGroup>
+  <Import Project="..\dir.props" />
+
+  <UsingTask AssemblyFile="$(TasksBinDir)Microsoft.DotNet.SourceBuild.Tasks.dll" TaskName="GetHostInformation" />
+  <UsingTask AssemblyFile="$(TasksBinDir)Microsoft.DotNet.SourceBuild.Tasks.dll" TaskName="BuildDummyPackages" />
+
+  <PropertyGroup>
+    <BuildCompetedSuccessSemaphore>$(BaseIntermediatePath)/init-build-proj.complete</BuildCompetedSuccessSemaphore>
+  </PropertyGroup>
+
+  <Target Name="Build"
+    Inputs="$(MSBuildThisFile)"
+    Outputs="$(BuildCompetedSuccessSemaphore)"
+    >
+    <CallTarget Targets="BuildTasks;WriteDynamicPropsToStaticPropsFiles;GenerateRootFs;ApplyPatches;BuildDummyPackages;BuildReferenceAssemblies" />
+    <Touch Files="$(BuildCompetedSuccessSemaphore)" AlwaysCreate="true" />
+  </Target>
+
+  <Target Name="Clean">
+    <Delete Files="$(BuildCompetedSuccessSemaphore)" />
+  </Target>
+
+  <Target Name="BuildTasks">
+    <Exec Command="$(DotNetCliToolDir)dotnet build tasks\Microsoft.DotNet.SourceBuild.Tasks\Microsoft.DotNet.SourceBuild.Tasks.csproj" />
+    <!-- Delete the packages directory to avoid polluting the repo builds -->
+    <RemoveDir Directories="$(ProjectDir)packages" />
```

Comment:
Do we have a property for this path?  If not, should we make one?

Reply:
We don't currently have a property for it from my quick look I was a little surprised but I also didn't see any other places where we use this path that could benefit from the property. At this point I'm unclear actually who is setting that path.

Reply:
The way it works today is that we are setting the NUGET_PACKAGES env var, and expecting everyone to respect it:

https://github.com/dotnet/source-build/blob/dev/release/2.0/build.sh#L16
https://github.com/dotnet/source-build/blob/dev/release/2.0/patches/corefx/0001-Respect-NUGET_PACKAGES-if-it-is-set.patch  (and others)

Reply:
That's write they are in the root scripts which is why we cannot easily share them. We should definitely working on moving that to the msbuild side of the house.